src/PerformanceSensitiveAnalyzers/Core/ExplicitAllocationAnalyzer.cs

```diff
@@ -83,17 +83,16 @@ protected override void AnalyzeNode(OperationAnalysisContext context, in Perform
                     return;
                 }
 
-                if (context.Operation.Parent.GetType().Name == "LazyConversionOperation")
+                if (context.Operation.Parent is IConversionOperation)
```

Comment:
💭 Will this fail for the following? Maybe add a new test?

```csharp
double x = new int();
```

Reply:
Yes, and also for implicit conversion between user-defined value types. Fixed and added two tests to avoid future regressions.

The ported diagnostic is still ObjectCreationRule though, which reports "Explicit new reference type allocation" on, for example, the following `object o = new int();`. It might be that this if clause can be completely removed once the TypeConversionAnalyzer is ported, so it might not be worth worrying about until we see if everything resolves nicely or not.